service/controller/app/resource/chartoperator/create_test.go

```diff
@@ -0,0 +1,250 @@
+				"3def4-kiam":               false,
+			},
+		},
+	}
+
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d: %s", i, tc.name), func(t *testing.T) {
+			var err error
+
+			schemeBuilder := runtime.SchemeBuilder{
+				v1alpha1.AddToScheme,
+			}
+
+			err = schemeBuilder.AddToScheme(scheme.Scheme)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+
+			var r *Resource
+			{
+				var objs []runtime.Object
+				objs = append(objs, tc.chartoperator)
+				for _, v := range tc.apps {
+					objs = append(objs, v)
+				}
+
+				fakeCtrlClient := fake.NewClientBuilder().
+					WithScheme(scheme.Scheme).
+					WithRuntimeObjects(objs...).
+					Build()
+
+				fakeK8sClient := clientgofake.NewSimpleClientset()
+				fakeLogger := microloggertest.New()
+
+				var valuesService *values.Values
+				{
+					c := values.Config{
+						K8sClient: fakeK8sClient,
+						Logger:    fakeLogger,
+					}
+
+					valuesService, err = values.New(c)
+					if err != nil {
+						t.Fatalf("error == %#v, want nil", err)
+					}
+				}
+
+				c := Config{
+					CtrlClient: fakeCtrlClient,
+					FileSystem: afero.NewMemMapFs(),
+					K8sClient:  fakeK8sClient,
+					Logger:     fakeLogger,
+
+					ChartNamespace:    "giantswarm",
+					WorkloadClusterID: "1abc2",
+					Values:            valuesService,
+				}
+
+				r, err = New(c)
+				if err != nil {
+					t.Fatalf("error == %#v, want nil", err)
+				}
+			}
+
+			var ctx context.Context
+			{
+				var objs []runtime.Object
+				for _, v := range tc.charts {
+					objs = append(objs, v)
+				}
+
+				fakeCtrlClient := fake.NewClientBuilder().
+					WithScheme(scheme.Scheme).
+					WithRuntimeObjects(objs...).
+					Build()
+
+				fakeClient := k8sclienttest.NewClients(k8sclienttest.ClientsConfig{
+					CtrlClient: fakeCtrlClient,
+					K8sClient:  clientgofake.NewSimpleClientset(),
+				})
+
+				c := controllercontext.Context{
+					Clients: controllercontext.Clients{
+						K8s: fakeClient,
+					},
+				}
+
+				ctx = controllercontext.NewContext(context.Background(), c)
+			}
+
+			err = r.triggerReconciliation(ctx, *tc.chartoperator)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+
+			var appList v1alpha1.AppList
+			err = r.ctrlClient.List(ctx, &appList)
+			if err != nil {
+				t.Fatalf("error == %#v, want nil", err)
+			}
+
+			for _, a := range appList.Items {
+				_, ok := a.GetAnnotations()[annotation.AppOperatorTriggerReconciliation]
+				expectedSet := tc.expectAnnotation[a.ObjectMeta.Name]
+				if expectedSet != ok {
+					t.Fatalf("%s: expected %t, got %t", a.ObjectMeta.Name, expectedSet, ok)
+				}
+
+				expectedNum := 1
+				if expectedSet {
+					expectedNum = 2
+				}
+
+				if expectedNum != len(a.GetAnnotations()) {
+					t.Fatalf("%s: expected %d, got %d", a.ObjectMeta.Name, expectedNum, len(a.GetAnnotations()))
+				}
+			}
+		})
+	}
+}
+
+func newV1alpha1App(crName, appName, cluster, organization string, inCluster bool) *v1alpha1.App {
```

Comment:
```suggestion
func newApp(crName, appName, cluster, organization string, inCluster bool) *v1alpha1.App {
```

Minor but I don't think we need the API version here.